File: src/main/java/org/droidmate/analyzer/exploration/IScenario.java
package org.droidmate.analyzer.exploration;

import org.droidmate.analyzer.api.IApi;

import java.nio.file.Path;
import java.util.List;

/**
 * Scenario (to be) executed during the experiment
 */
public interface IScenario {
    IExplorationResult getResult();

    void initialize();

    void setResult(IExplorationResult result);

    int getExplDepth();

    Path getCfgFile();

    Path getInlinedApk();

    void setInlinedApk(Path inlinedApk);

    List<IApi> getExploredApiList();

    List<IApi> getRestrictedApiList();

    boolean isValid();

    double getDissimilarity();
}

File: src/main/kotlin/org/droidmate/analyzer/exploration/IScenario.kt
package org.droidmate.analyzer.exploration

import org.droidmate.analyzer.api.IApi

import java.nio.file.Path

/**
 * Scenario (to be) executed during the experiment
 */
interface IScenario {
    var result: IExplorationResult?

    fun initialize()

    val explDepth: Int

    val cfgFile: Path

    var inlinedApk: Path

    val exploredApiList: List<IApi>

    val restrictedApiList: List<IApi>

    val isValid: Boolean

    val dissimilarity: Double
}
